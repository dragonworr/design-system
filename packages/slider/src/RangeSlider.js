import React from 'react'
import PropTypes from 'prop-types'
import Range from './rc-slider/Range'
import { space } from 'styled-system'
import styleSlider from './styleSlider'

const RangeSlider = styleSlider(Range)

RangeSlider.defaultProps = {
  allowCross: false,
  color: 'primary'
}

RangeSlider.propTypes = {
  ...space.propTypes,
  color: PropTypes.string
}

export default RangeSlider
