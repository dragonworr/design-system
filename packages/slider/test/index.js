import React from 'react'
import renderer from 'react-test-renderer'
import RangeSlider, { Slider } from '../src'

describe('pcln-slider', () => {
  test('RangeSlider renders', () => {
    const json = renderer.create(<RangeSlider />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('Slider renders', () => {
    const json = renderer.create(<Slider />).toJSON()
    expect(json).toMatchSnapshot()
  })
})
