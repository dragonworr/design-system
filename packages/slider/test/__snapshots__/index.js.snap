// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pcln-slider RangeSlider renders 1`] = `
.c0 {
  position: relative;
  height: 32px;
  padding-top: 12px;
  border-radius: 9999px;
  touch-action: none;
}

.c0 .rc-slider-rail,
.c0 .rc-slider-track {
  height: 8px;
}

.c0 .rc-slider-handle {
  width: 32px;
  height: 32px;
  margin-left: -16px;
  margin-top: -12px;
}

.c0 .rc-slider-rail {
  position: absolute;
  width: 100%;
  background-color: background.light;
  border-radius: 9999px;
}

.c0 .rc-slider-track {
  position: absolute;
  left: 0;
  border-radius: 9999px;
  background-color: primary;
}

.c0 .rc-slider-handle {
  position: absolute;
  cursor: pointer;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  border-radius: 9999px;
  border: solid 4px primary;
  background-color: background.lightest;
  touch-action: pan-x;
}

.c0 .rc-slider-handle:focus {
  box-shadow: 0 0 0 2px primary;
}

.c0.rc-slider-disabled {
  color: border.base;
}

.c0.rc-slider-disabled .rc-slider-handle {
  box-shadow: none;
  cursor: default;
}

<div
  className="rc-slider c0"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="rc-slider-rail"
    style={Object {}}
  />
  <div
    className="rc-slider-track rc-slider-track-1"
    style={
      Object {
        "left": "0%",
        "width": "0%",
      }
    }
  />
  <div
    className="rc-slider-step"
  />
  <div
    aria-disabled={false}
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="rc-slider-handle rc-slider-handle-1"
    onBlur={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="slider"
    style={
      Object {
        "left": "0%",
      }
    }
    tabIndex={0}
  />
  <div
    aria-disabled={false}
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="rc-slider-handle rc-slider-handle-2"
    onBlur={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="slider"
    style={
      Object {
        "left": "0%",
      }
    }
    tabIndex={0}
  />
  <div
    className="rc-slider-mark"
  />
</div>
`;

exports[`pcln-slider Slider renders 1`] = `
.c0 {
  position: relative;
  height: 32px;
  padding-top: 12px;
  border-radius: 9999px;
  touch-action: none;
}

.c0 .rc-slider-rail,
.c0 .rc-slider-track {
  height: 8px;
}

.c0 .rc-slider-handle {
  width: 32px;
  height: 32px;
  margin-left: -16px;
  margin-top: -12px;
}

.c0 .rc-slider-rail {
  position: absolute;
  width: 100%;
  background-color: background.light;
  border-radius: 9999px;
}

.c0 .rc-slider-track {
  position: absolute;
  left: 0;
  border-radius: 9999px;
  background-color: primary;
}

.c0 .rc-slider-handle {
  position: absolute;
  cursor: pointer;
  cursor: -webkit-grab;
  cursor: -moz-grab;
  cursor: grab;
  border-radius: 9999px;
  border: solid 4px primary;
  background-color: background.lightest;
  touch-action: pan-x;
}

.c0 .rc-slider-handle:focus {
  box-shadow: 0 0 0 2px primary;
}

.c0.rc-slider-disabled {
  color: border.base;
}

.c0.rc-slider-disabled .rc-slider-handle {
  box-shadow: none;
  cursor: default;
}

<div
  className="rc-slider c0"
  onBlur={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onMouseDown={[Function]}
  onMouseUp={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="rc-slider-rail"
    style={Object {}}
  />
  <div
    className="rc-slider-track"
    style={
      Object {
        "left": "0%",
        "width": "0%",
      }
    }
  />
  <div
    className="rc-slider-step"
  />
  <div
    aria-disabled={false}
    aria-valuemax={100}
    aria-valuemin={0}
    aria-valuenow={0}
    className="rc-slider-handle"
    onBlur={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="slider"
    style={
      Object {
        "left": "0%",
      }
    }
    tabIndex={0}
  />
  <div
    className="rc-slider-mark"
  />
</div>
`;
